### PR TITLE
Update jquery and related libraries

### PR DESCRIPTION
**Pull request by pbankonier**
## Description
Updated jquery to 3.3.1
Updated jqueryui to 1.12.1
Updated jquery validate to 1.17.0
Updated jquery flot to 0.8.3
Replace removed deprecated jquery functions
Fixe bugs that were introduced by the jquery update (mainly event handling)

## Types of changes
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?
Deployed this to our test environment and intensively tested UI functionalities by hand and checking the browser console for errors/warnings.

### CODE REVIEW

**Comment by svenvogel** [comment]
@pbankonier Great work!

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by svenvogel** [comment]
@rafaelweingartner can you check? 👍

**Comment by svenvogel** [comment]
@GabrielBrascher can you review too? 👍

**Comment by svenvogel** [comment]
@borisstoyanov we check and fix this. thanks for testing!

**Comment by borisstoyanov** [comment]
thanks @svenvogel, I'll try to fond some time today to spend with the UI and will report back any other failures.

**Comment by GabrielBrascher** [comment]
Thanks for the hard work @pbankonier! I will be testing it soon.

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by svenvogel** [comment]
@borisstoyanov @GabrielBrascher there are some checks not only in this pr which haven't completed... can you check the fix from @pbankonier ...

**Comment by borisstoyanov** [comment]
@blueorangutan package

File: ui/index.html [hunk 1]
```diff
@@ -102,7 +102,7 @@ <h3><translate key="label.select.a.zone"/></h3>
                                     <p><translate key="message.select.a.zone"/></p>
                                     <div class="select-area">
                                         <div class="desc"></div>
-                                        <select name="zoneid" class="required">
+                                        <select name="zoneid" class="required" required>
```
Comment (rafaelweingartner) [inline ui/index.html]: Is this a typo here?

**Comment by svenvogel** [comment]
@GabrielBrascher i agree with @borisstoyanov that most of these things weas seen in master before. we will look for the things to fix any of them but "Adding zone has dislocated buttons
image" seems a problem it was before and nobody fixed them. so maybe we will make a extra pr.

**Comment by GabrielBrascher** [comment]
@svenvogel @borisstoyanov, overall it looks good!

I marked as "requested changes" but I am ok with this one as it is; we can have another PR fixing those issues if they are not caused by this PR.

**Comment by GabrielBrascher** [comment]
What do you think of it @rafaelweingartner @borisstoyanov @rhtyd?

**Comment by rhtyd** [comment]
@GabrielBrascher I couldn't test this as it would require some manual setup and testing, can find some time after next week.

**Comment by GabrielBrascher** [comment]
I could not reproduce the same issues on the master (tested on firefox and chrome). Are you sure that the issues are from CloudStack master @svenvogel?

**Comment by svenvogel** [comment]
@GabrielBrascher we will Check this. You mean all issues?

You gave them good titles. I think „Adding zone has dislocated buttons“ is really a old problem. Please look #2787

Cheers

**Comment by rafaelweingartner** [comment]
@svenvogel can you guys try to use master (without your PR) to see if the same problem(dislocated buttons) appear there as well?

**Comment by svenvogel** [comment]
Here the other problem. same on master.

Autocomplete does not work when selecting rules for the domain admin

image
image

**Comment by rafaelweingartner** [comment]
About the second problem, I am not sure what can be causing it.
However, about the first one. Are you by any chance using a MAC OS?

**Comment by svenvogel** [comment]
yes i use macosx. i tried it with firefox 64.0.2 and chrome 71.0.3578.98. why do you ask?

**Comment by rafaelweingartner** [comment]
Can you try accessing the same Cloudstack that you have from a Windows or a Linux PC?

**Comment by svenvogel** [comment]
@rafaelweingartner we tried it also on a windows system. Windows 10 1803 with chrome 71.0.3578.98. we dont have a linux system but it should be clear that there is a problem.

**Comment by GabrielBrascher** [comment]
@svenvogel when using the master with Windows did you notice the same issues?

**Comment by GabrielBrascher** [comment]
@svenvogel I have been checking here and it indeed is happening on the master. This PR has already +2LGTMs, and tests have passed. It looks ready to merge. Those issues can be solved on another PR(s), thanks for the hard work guys!

**Pull request merged**